### PR TITLE
Prefill Name field when creating cases or studies

### PR DESCRIPTION
**Pull request by ayolab**
Signed-off-by: Ayoub LABIDI <ayoub.labidi@protonmail.com>

### CODE REVIEW

File: src/components/dialogs/create-study-dialog.js [hunk 1]
```diff
@@ -325,6 +328,16 @@ export const CreateStudyDialog = ({ open, onClose, providedCase }) => {
         }
     }, [activeDirectory]);
 
+    useEffect(() => {
+        if (studyNameRef.current.trim().length === 0 && selectedFile != null) {
```
Comment (etiennehomer) [inline src/components/dialogs/create-study-dialog.js]: Maybe you can mutualize this code

File: src/components/dialogs/field-hook.js [hunk 2]
```diff
@@ -54,6 +54,10 @@ export const useTextValue = ({
         setValue(event.target.value);
     }, []);
 
+    const handleExternalNameChange = useCallback((value) => {
```
Comment (etiennehomer) [inline src/components/dialogs/field-hook.js]: you can export setValue instead of creating this callback

**Review by etiennehomer — CHANGES_REQUESTED** [review]

**Comment by etiennehomer** [comment]
Maybe in this PR, we can use a fieldHook for studyName field

**Comment by ayolab** [comment]
I reused the nameField hook and the textValue and I created a custom hook for Name field pre-filling

**Comment by etiennehomer** [comment]
Tests OK

File: src/components/dialogs/field-hook.js [hunk 3]
```diff
@@ -250,9 +254,30 @@ export const useNameField = ({
...
+            selectedFile != null
+        ) {
+            setValue(
+                selectedFile.name.substr(0, selectedFile.name.indexOf('.'))
+            );
+        } else if (
+            selectedFile == null &&
+            setValue != null &&
+            setValue !== undefined
+        ) {
```
Comment (etiennehomer) [inline src/components/dialogs/field-hook.js]: just test && setValue

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/dialogs/field-hook.js (hunk 3 above)
Comment (etiennehomer) [reply src/components/dialogs/field-hook.js]: But your test on setValue should be upper nope ?
setValue is also used in the other branch of the if

Comment (ayolab) [reply src/components/dialogs/field-hook.js]: Done !

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**